### PR TITLE
[Triton] add onnx fp16 support

### PR DESCRIPTION
**Pull request by yuekaizhang**
This PR adds onnx fp16 support for streaming and offline ASR triton server. Also, it fixed several minor bugs. 

Currently triton support table:
| Mode  | Protobuf | Precision | Decoding Method | Model| Support |
| ------------- | ------------- | -------| ---------| -----| ----|
| Streaming  | Onnx  | FP32 | Greedy Search | Conformer-T| Yes|
| Offline  | Onnx  | FP32 | Greedy Search | Conformer-T| Yes|
| Streaming  | Onnx  | FP16 | Greedy Search | Conformer-T| Yes|
| Offline  | Onnx  | FP16 | Greedy Search | Conformer-T| Yes|
| Offline  | Jit  | FP32 | Greedy search| Conformer-T|Yes|

Would work on extending decoding methods and adding perf results in the future.

### CODE REVIEW

**Comment by yuekaizhang** [comment]
@csukuangfj  Hi, would you mind taking a look at this PR?

**Comment by csukuangfj** [comment]
> @csukuangfj Hi, would you mind taking a look at this PR?

Thanks!

File: triton/Dockerfile/Dockerfile.server [hunk 1]
```diff
@@ -6,8 +6,11 @@ LABEL maintainer="NVIDIA"
 LABEL repository="tritonserver"
 
 RUN pip3 install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu116
-RUN apt-get update  && apt-get -y install python3-dev && apt-get install -y cmake
-
+RUN apt-get update  && apt-get -y install python3-dev && apt-get install -y cmake && apt-get install -y libsndfile1
```
Comment (csukuangfj) [inline triton/Dockerfile/Dockerfile.server:9]: I think you can use a single command to install all of them, e.g,
```bash
apt-get install -y \
  python3-dev \
  cmake \
  libsndfile1
```

File: triton/model_repo_offline/greedy_search/1/model.py [hunk 2]
```diff
@@ -150,32 +149,34 @@ def execute(self, requests):
             start = offset
             end = offset + batch_size
             current_encoder_out = packed_encoder_out.data[start:end]
-            current_encoder_out = current_encoder_out.unsqueeze(1).unsqueeze(1).cpu().numpy()
-            # current_encoder_out's shape: (batch_size, 1, 1, encoder_out_dim)
+            #current_encoder_out = current_encoder_out.unsqueeze(1).unsqueeze(1).cpu().numpy()
```
Comment (csukuangfj) [inline triton/model_repo_offline/greedy_search/1/model.py:152]: Could you remove the unused code?

File: triton/model_repo_offline/greedy_search/1/model.py [hunk 3]
```diff
@@ -150,32 +149,34 @@ def execute(self, requests):
...
  
             offset = end
-
+        
             decoder_out = decoder_out[:batch_size]
            
-            in_joiner_tensor_0 = pb_utils.Tensor("encoder_out__0", current_encoder_out)
-            in_joiner_tensor_1 = pb_utils.Tensor("decoder_out__1", np.expand_dims(decoder_out, axis=1))
+            in_joiner_tensor_0 = pb_utils.Tensor("encoder_out", current_encoder_out)
+            #in_joiner_tensor_1 = pb_utils.Tensor("decoder_out", np.expand_dims(decoder_out, axis=1))
```
Comment (csukuangfj) [inline triton/model_repo_offline/greedy_search/1/model.py:161]: Also this one.

**Review by csukuangfj — COMMENTED** [review]
Thanks! Looks good to me.
Only left some minor comments.

**Comment by csukuangfj** [comment]
Thanks!

**Pull request merged**